src/components/PizzaList.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import "./PizzaList.scss";
+
+const selectUser = (reduxState) => {
+  return reduxState.user;
+};
+
+const selectPizzas = (reduxState) => {
+  return reduxState.pizzas.sort((a, b) => b.bought - a.bought);
+};
+
+export default function PizzaList() {
+  const user = useSelector(selectUser);
+  const pizzas = useSelector(selectPizzas);
+  const dispatch = useDispatch();
+  const [favorites, setFavorites] = useState(null);
+
+  useEffect(() => {
+    const allIngredients = {};
+    pizzas.forEach((pizza) => {
+      pizza.ingredients?.forEach((ingredient) => {
+        allIngredients[ingredient] = false;
+      });
+    });
```

Comment:
I see what you did here, its pretty smart! Now I ask you a question:
You are setting up an initial favourites object that has all the ingredients as keys and values false. Something like: ```{ tomatoe: false, muzzarella: false }```
What if instead you just used an array, where you would push the ingredient string in, so your favourites state would be: ```["tomatoe", "muzzarella"]``` or empty ```[]```

Reply:
This way:
1) you don't need to do any logic to prebuild it
2) your render becomes simpler, because you can do something like:
```
{ !favourites.length ? <render all pizzas> : pizzas.filter(p => p.ingredients.some(ing => favourites.includes(ing)) }
```

Reply:
Check out ```.some```, very useful array method! https://devdocs.io/javascript/global_objects/array/some
There is also ```.any``` that checks if **all** the elements fulfil the condition